src/yunohost/tools.py

```diff
@@ -565,3 +565,26 @@ def tools_diagnosis(auth, private=False):
         diagnosis['private']['domains'] = domain_list(auth)['domains']
 
     return diagnosis
+
+
+def tools_public_key():
+    key_type = 'ed25519'
```

Comment:
We had bad experience in YunoHost infra using this key format, it's apparently not really widly supported yet :/

This also should probably be a CLI argument.